src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -148,9 +159,7 @@ public IAfterCallSpecifiedConfiguration CallsBaseMethod()
 
             public IReturnValueConfiguration<TMember> WhenArgumentsMatch(Func<ArgumentCollection, bool> argumentsPredicate)
             {
-                Guard.AgainstNull(argumentsPredicate, nameof(argumentsPredicate));
-
-                this.ParentConfiguration.RuleBeingBuilt.UsePredicateToValidateArguments(argumentsPredicate);
+                this.ParentConfiguration.WhenArgumentsMatch(argumentsPredicate);

```

Comment:
Same as above, although it's not related to what I'm trying to fix in this PR (I can move it to a different PR if you prefer)